tests/kafkatest/services/kafka/kafka.py

```diff
@@ -89,11 +91,38 @@ def start(self):
                 topic_cfg["topic"] = topic
                 self.create_topic(topic_cfg)
 
+    def set_protocol_and_port(self, node):
```

Comment:
Looks better. I might just refactor the SASL tests back to using explicit ports per protocol. This wasn't really an option before as we couldn't run the sasl tests in virtual box, but I have a fix for that now. Although I don't really understand why this was changed in the first place. Maybe there is a good reason for doing it this way.